ipa-core/src/protocol/ipa_prf/quicksort.rs

```diff
@@ -134,7 +134,14 @@ where
     BitDecomposed<AdditiveShare<Boolean, SORT_CHUNK>>:
         for<'a> TransposeFrom<&'a [AdditiveShare<K>; SORT_CHUNK], Error = Infallible>,
 {
-    assert!(!ranges_to_sort.iter().any(Range::is_empty));
+    // Panic on any empty ranges in the input. Check if all the input ranges have length 1;
+    // if so, there's nothing to do.
+    if ranges_to_sort.iter().fold(true, |trivial, range| {
+        assert!(!range.is_empty());
+        trivial && range.len() == 1
+    }) {
+        return Ok(());
+    }
```

Comment:
I made the changes to quicksort first, before I added the special case in `oprf_ipa` that avoids calling it entirely. So the quicksort changes aren't essential. But given we're already checking for invalid ranges at the start of quicksort, the incremental cost of detecting the trivial case is minimal, and it lets us be more defensive in the rest of the function.